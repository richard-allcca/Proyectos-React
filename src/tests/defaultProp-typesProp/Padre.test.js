import { shallow } from "enzyme";
import React from "react";
// import { render } from "@testing-library/react";
import Padre from "../../conceptos_basicos/defaultProp-typesProp/Padre";

describe("Pruebas en  <Padre/>", () => {
  // ? testing con jest (render)
  // test("debe mostrar el mensaje Hola, Richard requerido!", () => {
  // const saludo = "hola, Richard requerido!";
  //=> sin destructurar
  // const wrapper = render(<Padre />);
  // wrapper.getByText()

  // => destructurado 'getByText'
  // const { getByText } = render(<Padre />);
  // expect(getByText(saludo)).toBeInTheDocument();
  // });

  // ? testing con enzyme (shallow)
  test("debe mostrar <Padre/> correctamente", () => {
    const wrapper = shallow(<Padre />);

    expect(wrapper).toMatchSnapshot();
  });
});

// Notas:
// toBeInTheDocument: valida que saludo existe en el document
