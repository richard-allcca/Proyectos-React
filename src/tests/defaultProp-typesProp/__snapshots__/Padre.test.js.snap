// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <Padre/> debe mostrar <Padre/> correctamente 1`] = `
<div>
  <Hijo
    numero={123}
    saludo="hola, Richard requerido!"
    subtitle="Soy un subTitulo"
  />
</div>
`;
